### DISCUSSION TITLE
OnChange when DefaultValues are set

### DISCUSSION THREAD

**Discussion by akiarostami** [react-hook-form/react-hook-form#2460] [General]
Hello,

I have a form with a set of DefaultValues. I am using OnChange to monitor the text change, but it only gets fired when a user types in something, not when the initial value is set using DefaultValues. Is there a way to have OnChange fired when DefaultValues are set as well? Or is there a different event I can use for that? 

Thank you.

**Comment by bluebill1049** [comment]
you mean `trigger`?

**Reply by akiarostami** [reply]
Yes

**Reply by bluebill1049** [reply]
`onChange={trigger('inputName'}` will that work for you? trigger individual field.

**Reply by akiarostami** [reply]
No, perhaps I didn't explain it well.

I have a counter that shows the remaining number of characters for an input (like Twitter). I am listening to OnChange to recalculate the number of characters, which works great.

The only issue is when there is a value assigned to <Input/> with RHF's DefaultValues, OnChange doesn't get fired (which is probably right). I have to separately check the default values and pass it separately to my component, which is ugly. I was wondering if there's any event to listen to for when there's an initial value, assigned by DefaultValues.

Thank you.

**Reply by bluebill1049** [reply]
oh i see! use `watch` and provide a default value that should solve this problem. and remove `onChange`

**Reply by akiarostami** [reply]
Thank you! This works. However, it seems that using watch slows things down significantly. Is this something you noticed in other places as well?

**Reply by bluebill1049** [reply]
you should watch a single input, not all of them :)

**Reply by akiarostami** [reply]
Haha... of course! I am only watching 2 fields, and see a significant performance change.

**Reply by bluebill1049** [reply]
if that's the case -> `useWatch`, you can isolate that re-render within your component level.

https://react-hook-form.com/api#useWatch

**Reply by akiarostami** [reply]
Thanks!
